c7n/resources/elasticbeanstalk.py

```diff
@@ -55,8 +68,161 @@ class resource_type(object):
         filter_name = 'EnvironmentNames'
         filter_type = 'list'
 
+    filter_registry = env_filters
+    action_registry = env_actions
+    retry = staticmethod(get_retry(('ThrottlingException',)))
+    permissions = ('elasticbeanstalk:ListTagsForResource',)
+
+    def augment(self, envs):
+        filter(None, _eb_env_tags(
+            envs, self.session_factory, self.executor_factory, self.retry
+        ))
+        return envs
+
+
+def _eb_env_tags(envs, session_factory, executor_factory, retry):
+    """Augment ElasticBeanstalk Environments with their tags."""
+
+    def process_tags(eb_env):
+        client = local_session(session_factory).client('elasticbeanstalk')
+        try:
+            tag_list = retry(
+                client.list_tags_for_resource,
+                ResourceArn=eb_env['EnvironmentArn']
+            )['ResourceTags']
+        except ClientError as e:
+            if e.response['Error']['Code'] == 'ResourceNotFoundException':
+                log.warning(
+                    "Exception getting elasticbeanstalk-environment tags\n %s",
+                    e
+                )
+            return None
+        eb_env['Tags'] = tag_list
+        return eb_env
+
+    # Handle API rate-limiting, which is a problem for accounts with many
+    # EB Environments
+    with executor_factory(max_workers=1) as w:
+        return list(w.map(process_tags, envs))
+
+
+@env_actions.register('mark-for-op')
+class TagDelayedAction(tags.TagDelayedAction):
+    """Mark an ElasticBeanstalk Environment for specific custodian action
+
+    Note that this will cause an update to the environment to deploy the tag
+    changes to all resources.
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: mark-for-delete
+                resource: elasticbeanstalk-environment
+                filters:
+                  - type: value
+                    key: CNAME
+                    op: regex
+                    value: .*inactive.*
+                actions:
+                  - type: mark-for-op
+                    op: terminate
+                    days: 7
+    """
+    schema = type_schema('mark-for-op', rinherit=tags.TagDelayedAction.schema)
+    permissions = ('elasticbeanstalk:UpdateTagsForResource',)
+
+    batch_size = 5
+
+    def process(self, envs):
+        return super(TagDelayedAction, self).process(envs)
+
+    def process_resource_set(self, envs, tags):
+        client = local_session(
+            self.manager.session_factory
+        ).client('elasticbeanstalk')
+        for env in envs:
+            client.update_tags_for_resource(
+                ResourceArn=env['EnvironmentArn'],
+                TagsToAdd=tags
+            )
+
+
+@env_actions.register('tag')
+@env_actions.register('mark')
```

Comment:
we're trying to avoid doing the aliases for new things, we  did it previously for compatibility reasons, but for new stuff we try to focus on the 'tag/remove-tag' nomenclature instead of the mark/unmark style.